### PR TITLE
Enforce virtual file type when it is falsely marked as non-virtual.  Added more logs for scenarios when virtual placeholder gets marked as non-virtual.

### PR DESCRIPTION
**Pull request by allexzander**

### CODE REVIEW

File: src/libsync/discovery.cpp [hunk 1]
```diff
@@ -498,6 +498,7 @@ void ProcessDirectoryJob::processFile(PathTuple path,
 
     if (dbEntry._modtime == localEntry.modtime && dbEntry._type == ItemTypeVirtualFile && localEntry.type == ItemTypeFile) {
         item->_type = ItemTypeFile;
+        qCWarning(lcDisco) << "Changing item type from virtual to normal file";
```
Comment (mgallien) [inline src/libsync/discovery.cpp:501]: general question
why warning level and not just info level ?
I would tend to think that those code decisions however strange they seem to be are probably done on purpose (that was my understanding)

File: src/libsync/syncfileitem.cpp [hunk 2]
```diff
@@ -81,8 +81,10 @@ SyncJournalFileRecord SyncFileItem::toSyncJournalFileRecordWithInode(const QStri
 
     // Some types should never be written to the database when propagation completes
     rec._type = _type;
-    if (rec._type == ItemTypeVirtualFileDownload)
+    if (rec._type == ItemTypeVirtualFileDownload) {
         rec._type = ItemTypeFile;
+        qCWarning(lcFileItem) << "Changing item type from virtual to normal file";
```
Comment (mgallien) [inline src/libsync/syncfileitem.cpp:86]: maybe  also add the content of the comment a few lines before to the debug output
would probably speed up any future analyze of logs

File: src/libsync/vfs/cfapi/vfs_cfapi.cpp [hunk 3]
```diff
@@ -396,8 +396,19 @@ void VfsCfApi::requestHydration(const QString &requestId, const QString &path)
...
+            qCWarning(lcCfApi) << "Hydration requested for a placeholder file not marked as virtual in local DB. Attempting to fix it...";
+            record._type = ItemTypeVirtualFileDownload;
+            isNotVirtualFileFailure = !journal->setFileRecord(record);
+        } else {
+            isNotVirtualFileFailure = true;
+        }
+    }
+
+    if (isNotVirtualFileFailure) {
+        qCWarning(lcCfApi) << "Couldn't hydrate, the file is not virtual";
```
Comment (mgallien) [inline src/libsync/vfs/cfapi/vfs_cfapi.cpp:411]: seems to me something I could reproduce
like doing those steps:

1. implicitly hydrate a very big file
2. override it with a real file during the hydration
3. check that you get an error

File: src/libsync/syncfileitem.cpp (hunk 2 above)
Comment (mgallien) [reply src/libsync/syncfileitem.cpp]: replied on the new version of the code

File: src/libsync/vfs/cfapi/vfs_cfapi.cpp (hunk 3 above)
Comment (allexzander) [reply src/libsync/vfs/cfapi/vfs_cfapi.cpp:411]: @mgallien 
yes, but, then I am able to hydrate that file again after closing the popup or restarting the desktop client, so not the same as this issue, the file is not hydrated in this case but it is online-only placeholder

**Review by mgallien — APPROVED** [review]

**Review by claucambra — APPROVED** [review]

**Pull request merged**